com.civicrm.hrjobroles/js/src/job-roles/services/date-validation.js

```diff
@@ -0,0 +1,78 @@
+define([
+    'job-roles/services/services',
+    'common/moment'
+], function (module, moment) {
+    /**
+     * Service responsible for validating dates in HRJobRoles
+     * @constructor
+     */
+    module.service('DateValidation', ['HR_settings', function (HR_settings) {
+        var me = this;
+
+        me._error = function (error_msg, fields) {
+            throw new Error(error_msg, fields[0]);
+        };
+
+        me.defaultFormat = HR_settings.DATE_FORMAT;
+
+        /**
+         *
+         * @param {function} error
+         * @throws TypeError
+         */
+        me.setErrorCallback = function setErrorCallback(error) {
+            if (typeof error === 'function') {
+                me._error = error;
+            } else {
+                throw new TypeError('Error callback must be a function.');
+            }
+        };
+
+        me.dateFormats = [
+            'x',
+            'YYYY-MM-DD'
+        ];
+
+        me.defaultFormat && me.dateFormats.push(me.defaultFormat.toUpperCase());
+
+        me.validate = function validate(start, end) {
```

Comment:
Please document these functions. For example here it seems `start` and `end` can be of different type, so we need to document it